libs/sim/simulator.rs

```diff
@@ -162,14 +164,20 @@ impl<'a> Simulator {
             .expect("failed to execute process");
     }
 
-    pub fn tick(&mut self, tick: u64) {
+    // if self.record_network_data is set, returns amt of data used per
+    // tick
+    pub fn tick(&mut self, tick: u64) -> usize {
         log::info!("################# TICK {0} START #################", tick);
         let mut rpc_buffer = vec![];
+        let mut data_over_network_size = 0;
         // tick all elements to generate RPCs
         // this is the send phase. collect all the RPCs
         for (_elem_name, element_obj) in self.elements.iter_mut() {
             let rpcs = element_obj.tick(tick);
             for rpc in &rpcs {
+                if self.record_network_data {
+                    data_over_network_size += rpc.len();
+                }
```

Comment:
You should do this with the rpc buffer instead,  not in the loop and only when the check is enabled.